plugin/proto.go

```diff
@@ -0,0 +1,56 @@
+// Package plugin defines the protocol layer for communication between notation
+// and notation external plugin.
+package plugin
```

Comment:
I guess I still have the question, why do we need the duplicate implementation here? Can we reuse https://github.com/notaryproject/notation-go/tree/v1.0.0/log and https://github.com/notaryproject/notation-go/tree/v1.0.0/plugin for now? And when we want to refactor in the future, we could make the changes.

Reply:
The plugin doesn't need most of the dependencies which notation-go directly or indirectly has. Also, most probably plugin producer and consumer won't like it.